### DISCUSSION TITLE
wagmi v2: disconnect not working (immediately reconnects with another connector)

### DISCUSSION THREAD

**Discussion by kristiegiles** [wevm/wagmi#3537] [Q&A]
Sometimes when I call `disconnectAsync`, it immediately reconnects with another automatically injected [eip-6963](https://eips.ethereum.org/EIPS/eip-6963) connector. From a user perspective, it appears that the disconnect button is not doing anything, but really it's switching connectors. Anyone have thoughts on the best way to deal with this? 
![Screenshot 2024-01-30 at 5 08 53 PM](https://github.com/wevm/wagmi/assets/25679998/98877aab-9038-4262-85bf-4b8328bad7a9)

Also, I sometimes get duplicate connectors, which seems unexpected. I assume this may be an issue on the MetaMask side? I guess I should dedupe them manually? I am not using the metaMaskSDK connector because of the [undefined mobile provider issue](https://github.com/wevm/wagmi/issues/3400). 
<img width="347" alt="Screenshot 2024-01-30 at 3 13 41 PM" src="https://github.com/wevm/wagmi/assets/25679998/71bbdf35-b9c6-468a-a5af-676b30c895d6">
<img width="1044" alt="Screenshot 2024-01-30 at 3 14 44 PM" src="https://github.com/wevm/wagmi/assets/25679998/b3b24370-ca4c-42a4-a956-3d01bdc80d81">

**Comment by kristiegiles** [comment]
I ended up getting the duplicate eip-6963 connectors with other wallets besides MetaMask.

**Comment by tmm** [comment]
When you disconnect, Wagmi [switches to the next connection](https://github.com/wevm/wagmi/blob/main/packages/core/src/actions/disconnect.ts#L55) if there are any.

**Reply by kristiegiles** [reply]
Thanks. Would the recommendation be to not use the injected connector, metamask connector, etc.? So only the automatically injected connector will work?

**Reply by tmm** [reply]
It depends on what you are trying to do with your app. Between the [internal EIP-6963 support](https://wagmi.sh/core/api/createConfig#multiinjectedproviderdiscovery), and [`coinbaseWallet`](https://wagmi.sh/react/api/connectors/coinbaseWallet) and [`walletConnect`](https://wagmi.sh/react/api/connectors/walletConnect) connectors, most wallets should be covered.
